sympy/simplify/tests/test_cse.py

```diff
@@ -412,13 +412,15 @@ def test_issue_7840():
 
 
 def test_issue_8891():
-    for cls in (MutableDenseMatrix, MutableSparseMatrix,
-            ImmutableDenseMatrix, ImmutableSparseMatrix):
-        m = cls(2, 2, [x + y, 0, 0, 0])
+    for cls_input, cls_output in ((MutableDenseMatrix, ImmutableDenseMatrix),
+            (MutableSparseMatrix, ImmutableSparseMatrix),
+            (ImmutableDenseMatrix, ImmutableDenseMatrix),
+            (ImmutableSparseMatrix, ImmutableSparseMatrix)):
```

Comment:
This is a minor regression too. I'm not sure if it's a big deal, though. @bjodah, @moorepants, others, how do you feel about cse on a mutable matrix returning an immutable matrix?

Reply:
Not real sure on the potential consequences but it would be nice if cse always returned what you passed in.

Reply:
I agree, if there's no way around it: then it's fine. Otherwise I'd also prefer keeping the original type.

Reply:
@mmohrhard Is there a way to retain the type?